### DISCUSSION TITLE
Add support for more parameters when using oauth2

### DISCUSSION THREAD

**Discussion by rvangsgaard** [redpanda-data/connect#2049] [Ideas]
In order for oauth2 to succeed in our Azure AD setup, we must specify the `resource` parameter when getting bearer token. Other users might have the same issue.

I have tried to add the capability of adding payload parameters in the "get token" request in a PR https://github.com/benthosdev/benthos/pull/2047

**Comment by devantler** [comment]
I found a few issues related to the lack of payload parameters for the OAuth2 request:

#1812 
#1213 

I hope this might raise awareness as it currently limits the possibility of using Benthos. 🙏🏻
